cnc_ctrl_v1/GCode.cpp

```diff
@@ -370,6 +370,56 @@ byte  executeBcodeLine(const String& gcodeLine){
         return STATUS_OK;
     }
 
+    if(gcodeLine.substring(0, 3) == "B17"){
+        //The B17 sets the current Z position as Upper Limit for zAxis
+
+        sysSettings.zAxisUpperLimit = zAxis.read();
+        settingsSaveToEEprom();
+
+        Serial.print("Upper limit set to: ");
+        Serial.print(isnan(sysSettings.zAxisUpperLimit) ? "   ": String(sysSettings.zAxisUpperLimit / sys.inchesToMMConversion));
+        Serial.println(" ");
+        
+    
+        return STATUS_OK;
+    }
+ 
+       if(gcodeLine.substring(0, 3) == "B18"){
+        //The B18 sets the current Z position as Lower Limit for zAxis
+
+        sysSettings.zAxisLowerLimit = zAxis.read();
+        settingsSaveToEEprom();
+
+        Serial.print("Lower limit set to: ");
+        Serial.print(isnan(sysSettings.zAxisLowerLimit) ? "   ": String(sysSettings.zAxisLowerLimit / sys.inchesToMMConversion));
+        Serial.println(" ");        
+    
+        return STATUS_OK;
+    }
+       if(gcodeLine.substring(0, 3) == "B19"){
+        //The B19 clears limits for zAxis
+
+        sysSettings.zAxisUpperLimit = NAN;
+        sysSettings.zAxisLowerLimit = NAN;
+        settingsSaveToEEprom();
```

Comment:
It needs to update eeprom like any other non volatile Maslow setting, otherwise those values will be back on program restart.

eeprom has a minimun life of 100,000 write/re-write cycles.   that's plenty for this use.